### DISCUSSION TITLE
Exclude a specific doc category based on different environments?

### DISCUSSION THREAD

**Discussion by sparanoid** [facebook/docusaurus#5181] [Q&A] [answered]
Hi all, I'd like to know if there's a easy way to exclude a specific doc category based on different environments. The most related feature seems to be the pages plugin that can exclude pages with patterns.

My use case is to hide categories administration, examples, and etc on prod, dev environments. But to make them show on demo and office environments.

My workaound at the moment is creating different sidebars based on environments to "hide" them from the site. But the docs hidden from sidebar are still generated and accessible via URLs. I'd like to know if there's an elegant way to trully exclude a category based on different environments?

Thanks!

**Comment by slorber** [comment, marked as answer]
This is something some Facebook sites are doing: filter the sidebars.js file with your own logic:
https://github.com/facebook/relay/blob/main/website/sidebars.js

You can run any nodejs code there, including reading `process.env`

However I don't know how they do to make the URLs unavailable, maybe the docs are simply not added to the docs folder?

This PR adds an `exclude` config to all content plugins, it might help you achieve what you want: https://github.com/facebook/docusaurus/pull/5173

**Reply by slorber** [reply]
cc @jknoxville any hint to share?

**Reply by jknoxville** [reply]
Yes sadly the way we do it inside Facebook probably isn't going to help here.

We use env variables to add extra internal only pages to the sidebar, this variant of the site is built and hosted internally.

The website files are synced with GitHub repositories, but the internal page files are excluded from that sync, so they don't exist in the GitHub repo when it's building the public variant of the site.

**Reply by slorber** [reply]
Thanks

Even if they existed in the repo you could probably exclude them with the new exclude config coming in beta.4:

https://docusaurus.io/docs/next/api/plugins/@docusaurus/plugin-content-docs

```js
        exclude: [
           '**/fbInternal/**',


          '**/_*.{js,jsx,ts,tsx,md,mdx}',
          '**/_*/**',
          '**/*.test.{js,jsx,ts,tsx}',
          '**/__tests__/**',
        ],
```

**Reply by sparanoid** [reply]
@slorber Thanks, I tried latest canary release but got some issues. I guess when the working directory contains underscore the build process will throw [`MDX partial should not contain FrontMatter`](https://github.com/facebook/docusaurus/pull/5173/files#diff-3cb772801689197b1f332b6d0938de44c3ed1626583bdd7b6c87b0fef775315fR94) error.

This is reproducible with [GitHub self-hosted runner](https://docs.github.com/en/actions/hosting-your-own-runners/about-self-hosted-runners):

```log
...
2021-07-18T06:58:54.4570148Z Working directory is '/home/ubuntu/actions-runner/_work/openbayes-docs/openbayes-docs'

...

2021-07-18T06:59:24.0450511Z [success] [webpackbar] Client: Compiled with some errors in 16.16s
2021-07-18T06:59:24.0529511Z 
2021-07-18T06:59:24.0530401Z 
2021-07-18T06:59:24.0532146Z Error: MDX partial should not contain FrontMatter: /home/ubuntu/actions-runner/_work/openbayes-docs/openbayes-docs/docs/admin/admin.md
2021-07-18T06:59:24.0534117Z Error: MDX partial should not contain FrontMatter: /home/ubuntu/actions-runner/_work/openbayes-docs/openbayes-docs/docs/admin/internal-ssh.md
2021-07-18T06:59:24.0535181Z     at runMicrotasks (<anonymous>)
```

While the offical runner works fine:

```
...
2021-07-18T06:47:13.9136221Z Working directory is '/home/runner/work/openbayes-docs/openbayes-docs'

...

2021-07-18T06:48:13.7116076Z [success] [webpackbar] Client: Compiled successfully in 32.21s
2021-07-18T06:48:19.0822996Z [success] [webpackbar] Server: Compiled successfully in 37.56s
2021-07-18T06:48:29.5291392Z Success! Generated static files in "build/docs".
```

**Reply by slorber** [reply]
@sparanoid if you already have existing docs in a folder starting with _, they'll be excluded by default. 
Those files can still be used, but only as "partials", imported by mdx in other mdx files. 
And those files should not contain frontmatter because that frontmatter will not be used by Docusaurus.

But you are right, it seems there is a bug because `_work` is a folder outside of your Docusaurus site, so it shouldn't have any effect on Docusauurs, will get this fixed!

**Reply by slorber** [reply]
@sparanoid watch https://github.com/facebook/docusaurus/pull/5199 for the fix
